pkg/controller/datavolume/clone-controller-base.go

```diff
@@ -426,7 +426,7 @@ var populatorPhaseMap = map[string]cdiv1.DataVolumePhase{
 	clone.RebindPhaseName:        cdiv1.RebindInProgress,
 	clone.SnapshotClonePhaseName: cdiv1.CloneFromSnapshotSourceInProgress,
 	clone.SnapshotPhaseName:      cdiv1.SnapshotForSmartCloneInProgress,
-	//clone.ErrorPhaseName:         cdiv1.Error, // Want to hold off on this for now
+	clone.ErrorPhaseName:         cdiv1.Failed,
```

Comment:
Sorry but I really didnt understand this change and commit and how 'unknown' is related to this change..
Also if we do get to Error phase in the clone, with this change nothing will be shown in the dv?

Reply:
We use this map as a chart to get the appropriate DV phase depending on the AnnClonePhase annotation. We didn't have anything for error so we returned empty and the case was handled as an unknown populator phase: https://github.com/kubevirt/containerized-data-importer/blob/e7e0586e0cd3d37c90baafe8dcd48565371b3805/pkg/controller/datavolume/clone-controller-base.go#L436

This change does not affect the DV phase because we never set it to `Failed` when a clone fails, but it allows us to record an event to the DV that could help debug (check `updateStatusPhaseForPopulator`).

Reply:
so Error phase is a terminal phase? cant come back from it? how come we have it in populator but not in dv?
Why was it a problem to have error phase in dv? I would have preferred to have some indication also in the dv status..

Reply:
We might be able to come back for an error phase, that's why we are cautious and avoid setting the DV to `Failed` unless we are sure it's terminal. We use annotations for populator phases and that's different than an object phase, but still may need some improvements. 

> Why was it a problem to have error phase in dv? 

We usually try to avoid it unless we are 100% sure it's terminal. However, this PR is just trying to improve that behavior without being too intrusive. With populators, we've been trying to be consistent with non-populator DV phases from the beginning.

Reply:
Sorry I still dont understand why not do:
`clone.ErrorPhaseName:         cdiv1.Error`
and have this updated in the dv? how is this any different then any other populator specific phase?
and lets say we dont want to update in the dv why not do 
`if dvPhase != cdiv1.Error {`
How is Failed related to all this? as you said error!=failed